glog shadow breaks transitive builds.

<!-- This form is for bug reports and feature requests ONLY!
Also make sure that you visit our User Guide at https://kubevirt.io/user-guide/
-->

**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind enhancement


**What happened**:
the glog build shadowing glog -> ../../../pkg/staging/glog//

breaks transitive builds, if they have libs that actually depend on glog. 

The whole dependency closure doesn't work. 

```
vendor/kubevirt.io/kubevirt/pkg/log/log.go:58:24: undefined: glog.LogLevel
vendor/kubevirt.io/kubevirt/pkg/log/log.go:59:24: undefined: glog.LogLevel
Makefile:39: recipe for target 'vet' failed

```

**What you expected to happen**:

For kubevirt to not shadow glog.

**How to reproduce it (as minimally and precisely as possible)**:

Copy the  cmd/example-hook-sidecar folder into an empty folder. 

copy this Gopkg.toml 

https://gist.github.com/43f0fddf5876eb26f510d16c71923ead

and try building 

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`):

Latest head as of today - see Gopkg.toml for reference

--- a/cmd/fake-qemu-process/fake-qemu.go
+++ b/cmd/fake-qemu-process/fake-qemu.go
@@ -20,7 +20,6 @@
 package main
 
 import (
-	"flag"
 	"fmt"
 	"os"
 	"os/signal"
@@ -31,13 +30,12 @@ import (
 )
 
 func main() {
-	uuid := flag.String("uuid", "", "some fake arg")
+	uuid := pflag.String("uuid", "", "some fake arg")
 	c := make(chan os.Signal, 1)
 	signal.Notify(c, os.Interrupt,
 		syscall.SIGTERM,
 	)
 
-	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
 	pflag.Parse()
 	fmt.Printf("Started fake qemu process with uuid %s\n", *uuid)
 

--- a/cmd/virt-handler/virt-handler.go
+++ b/cmd/virt-handler/virt-handler.go
@@ -99,7 +99,7 @@ func (app *virtHandlerApp) Run() {
 	}
 
 	logger := log.Log
-	logger.V(1).Level(glog.INFO).Log("hostname", app.HostOverride)
+	logger.V(1).Level(log.INFO).Log("hostname", app.HostOverride)
 
 	// Create event recorder
 	virtCli, err := kubecli.GetKubevirtClient()

--- a/cmd/virt-launcher/virt-launcher.go
+++ b/cmd/virt-launcher/virt-launcher.go
@@ -20,7 +20,7 @@
 package main
 
 import (
-	"flag"
+	goflag "flag"
 	"fmt"
 	"os"
 	"os/exec"
@@ -303,23 +303,23 @@ func waitForFinalNotify(deleteNotificationSent chan watch.Event,
 }
 
 func main() {
-	qemuTimeout := flag.Duration("qemu-timeout", defaultStartTimeout, "Amount of time to wait for qemu")
-	virtShareDir := flag.String("kubevirt-share-dir", "/var/run/kubevirt", "Shared directory between virt-handler and virt-launcher")
-	ephemeralDiskDir := flag.String("ephemeral-disk-dir", "/var/run/kubevirt-ephemeral-disks", "Base directory for ephemeral disk data")
-	name := flag.String("name", "", "Name of the VirtualMachineInstance")
-	uid := flag.String("uid", "", "UID of the VirtualMachineInstance")
-	namespace := flag.String("namespace", "", "Namespace of the VirtualMachineInstance")
-	watchdogInterval := flag.Duration("watchdog-update-interval", defaultWatchdogInterval, "Interval at which watchdog file should be updated")
-	readinessFile := flag.String("readiness-file", "/tmp/health", "Pod looks for this file to determine when virt-launcher is initialized")
-	gracePeriodSeconds := flag.Int("grace-period-seconds", 30, "Grace period to observe before sending SIGTERM to vm process")
-	useEmulation := flag.Bool("use-emulation", false, "Use software emulation")
-	hookSidecars := flag.Uint("hook-sidecars", 0, "Number of requested hook sidecars, virt-launcher will wait for all of them to become available")
-	noFork := flag.Bool("no-fork", false, "Fork and let virt-launcher watch itself to react to crashes if set to false")
+	qemuTimeout := pflag.Duration("qemu-timeout", defaultStartTimeout, "Amount of time to wait for qemu")
+	virtShareDir := pflag.String("kubevirt-share-dir", "/var/run/kubevirt", "Shared directory between virt-handler and virt-launcher")
+	ephemeralDiskDir := pflag.String("ephemeral-disk-dir", "/var/run/kubevirt-ephemeral-disks", "Base directory for ephemeral disk data")
+	name := pflag.String("name", "", "Name of the VirtualMachineInstance")
+	uid := pflag.String("uid", "", "UID of the VirtualMachineInstance")
+	namespace := pflag.String("namespace", "", "Namespace of the VirtualMachineInstance")
+	watchdogInterval := pflag.Duration("watchdog-update-interval", defaultWatchdogInterval, "Interval at which watchdog file should be updated")
+	readinessFile := pflag.String("readiness-file", "/tmp/health", "Pod looks for this file to determine when virt-launcher is initialized")
+	gracePeriodSeconds := pflag.Int("grace-period-seconds", 30, "Grace period to observe before sending SIGTERM to vm process")
+	useEmulation := pflag.Bool("use-emulation", false, "Use software emulation")
+	hookSidecars := pflag.Uint("hook-sidecars", 0, "Number of requested hook sidecars, virt-launcher will wait for all of them to become available")
+	noFork := pflag.Bool("no-fork", false, "Fork and let virt-launcher watch itself to react to crashes if set to false")
 
 	// set new default verbosity, was set to 0 by glog
-	flag.Set("v", "2")
+	goflag.Set("v", "2")
 
-	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+	pflag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("v"))
 	pflag.Parse()
 
 	log.InitializeLogging("virt-launcher")

--- a/hack/dep-prune.sh
+++ b/hack/dep-prune.sh
@@ -1,2 +1,5 @@
 #!/bin/bash
 for file in $(find vendor/ -name "*_test.go"); do rm ${file}; done
+rm -rf vendor/github.com/golang/glog
+mkdir -p vendor/github.com/golang/
+ln -s ../../../pkg/staging/glog/ vendor/github.com/golang/glog

--- a/pkg/controller/controller.go
+++ b/pkg/controller/controller.go
@@ -23,7 +23,6 @@ import (
 	"fmt"
 	"runtime/debug"
 
-	"github.com/golang/glog"
 	k8sv1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/fields"
@@ -71,7 +70,7 @@ func NewListWatchFromClient(c cache.Getter, resource string, namespace string, f
 
 func HandlePanic() {
 	if r := recover(); r != nil {
-		log.Log.Level(glog.FATAL).Log("stacktrace", debug.Stack(), "msg", r)
+		log.Log.Level(log.FATAL).Log("stacktrace", debug.Stack(), "msg", r)
 	}
 }
 

--- a/pkg/log/log.go
+++ b/pkg/log/log.go
@@ -21,6 +21,7 @@ package log
 
 import (
 	"errors"
+	goflag "flag"
 	"fmt"
 	"io"
 	"os"
@@ -44,6 +45,22 @@ const (
 	libvirtTimestampFormat = "2006-01-02 15:04:05.999-0700"
 )
 
+type LogLevel int32
+
+const (
+	INFO LogLevel = iota
+	WARNING
+	ERROR
+	FATAL
+)
+
+var LogLevelNames = map[LogLevel]string{
+	INFO:    "info",
+	WARNING: "warning",
+	ERROR:   "error",
+	FATAL:   "fatal",
+}
+
 var lock sync.Mutex
 
 type LoggableObject interface {
@@ -54,8 +71,8 @@ type LoggableObject interface {
 type FilteredLogger struct {
 	logContext            *log.Context
 	component             string
-	filterLevel           glog.LogLevel
-	currentLogLevel       glog.LogLevel
+	filterLevel           LogLevel
+	currentLogLevel       LogLevel
 	verbosityLevel        int
 	currentVerbosityLevel int
 	err                   error
@@ -66,7 +83,9 @@ var Log = DefaultLogger()
 func InitializeLogging(comp string) {
 	defaultComponent = comp
 	Log = DefaultLogger()
-	glog.Setup(comp, getDefaultVerbosity())
+	glog.CopyStandardLogTo(LogLevelNames[INFO])
+	goflag.CommandLine.Set("component", comp)
+	goflag.CommandLine.Set("logtostderr", "true")
 }
 
 func getDefaultVerbosity() int {
@@ -82,7 +101,7 @@ func getDefaultVerbosity() int {
 
 // Wrap a go-kit logger in a FilteredLogger. Not cached
 func MakeLogger(logger log.Logger) *FilteredLogger {
-	defaultLogLevel := glog.INFO
+	defaultLogLevel := INFO
 
 	defaultVerbosity = getDefaultVerbosity()
 	// This verbosity will be used for info logs without setting a custom verbosity level
@@ -130,9 +149,11 @@ func DefaultLogger() *FilteredLogger {
 	return Logger(defaultComponent)
 }
 
+// SetIOWriter is meant to be used for testing. "log" and "glog" logs are sent to /dev/nil.
+// KubeVirt related log messages will be sent to this writer
 func (l *FilteredLogger) SetIOWriter(w io.Writer) {
 	l.logContext = log.NewContext(log.NewJSONLogger(w))
-	glog.SetIOWriter(w)
+	goflag.CommandLine.Set("logtostderr", "false")
 }
 
 func (l *FilteredLogger) SetLogger(logger log.Logger) *FilteredLogger {
@@ -164,7 +185,7 @@ func (l FilteredLogger) log(skipFrames int, params ...interface{}) error {
 	// messages should be logged if any of these conditions are met:
 	// The log filtering level is info and verbosity checks match
 	// The log message priority is warning or higher
-	if l.currentLogLevel >= glog.WARNING || (l.filterLevel == glog.INFO &&
+	if l.currentLogLevel >= WARNING || (l.filterLevel == INFO &&
 		(l.currentLogLevel == l.filterLevel) &&
 		(l.currentVerbosityLevel <= l.verbosityLevel)) {
 		now := time.Now().UTC()
@@ -172,7 +193,7 @@ func (l FilteredLogger) log(skipFrames int, params ...interface{}) error {
 		logParams := make([]interface{}, 0, 8)
 
 		logParams = append(logParams,
-			"level", glog.LogLevelNames[l.currentLogLevel],
+			"level", LogLevelNames[l.currentLogLevel],
 			"timestamp", now.Format("2006-01-02T15:04:05.000000Z"),
 			"pos", fmt.Sprintf("%s:%d", filepath.Base(fileName), lineNumber),
 			"component", l.component,
@@ -250,8 +271,8 @@ func (l *FilteredLogger) WithPrefix(obj ...interface{}) *FilteredLogger {
 	return l
 }
 
-func (l *FilteredLogger) SetLogLevel(filterLevel glog.LogLevel) error {
-	if (filterLevel >= glog.INFO) && (filterLevel <= glog.FATAL) {
+func (l *FilteredLogger) SetLogLevel(filterLevel LogLevel) error {
+	if (filterLevel >= INFO) && (filterLevel <= FATAL) {
 		l.filterLevel = filterLevel
 		return nil
 	}
@@ -281,41 +302,41 @@ func (l FilteredLogger) Reason(err error) *FilteredLogger {
 	return &l
 }
 
-func (l FilteredLogger) Level(level glog.LogLevel) *FilteredLogger {
+func (l FilteredLogger) Level(level LogLevel) *FilteredLogger {
 	l.currentLogLevel = level
 	return &l
 }
 
 func (l FilteredLogger) Info(msg string) {
-	l.Level(glog.INFO).msg(msg)
+	l.Level(INFO).msg(msg)
 }
 
 func (l FilteredLogger) Infof(msg string, args ...interface{}) {
-	l.Level(glog.INFO).msgf(msg, args...)
+	l.Level(INFO).msgf(msg, args...)
 }
 
 func (l FilteredLogger) Warning(msg string) {
-	l.Level(glog.WARNING).msg(msg)
+	l.Level(WARNING).msg(msg)
 }
 
 func (l FilteredLogger) Warningf(msg string, args ...interface{}) {
-	l.Level(glog.WARNING).msgf(msg, args...)
+	l.Level(WARNING).msgf(msg, args...)
 }
 
 func (l FilteredLogger) Error(msg string) {
-	l.Level(glog.ERROR).msg(msg)
+	l.Level(ERROR).msg(msg)
 }
 
 func (l FilteredLogger) Errorf(msg string, args ...interface{}) {
-	l.Level(glog.ERROR).msgf(msg, args...)
+	l.Level(ERROR).msgf(msg, args...)
 }
 
 func (l FilteredLogger) Critical(msg string) {
-	l.Level(glog.FATAL).msg(msg)
+	l.Level(FATAL).msg(msg)
 }
 
 func (l FilteredLogger) Criticalf(msg string, args ...interface{}) {
-	l.Level(glog.FATAL).msgf(msg, args...)
+	l.Level(FATAL).msgf(msg, args...)
 }
 
 func LogLibvirtLogLine(logger *FilteredLogger, line string) {

--- a/pkg/log/log_test.go
+++ b/pkg/log/log_test.go
@@ -27,7 +27,6 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/golang/glog"
 	k8sv1 "k8s.io/api/core/v1"
 	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
@@ -96,7 +95,7 @@ func TestBadLevel(t *testing.T) {
 func TestGoodLevel(t *testing.T) {
 	setUp()
 	l := Logger("test")
-	error := l.SetLogLevel(glog.INFO)
+	error := l.SetLogLevel(INFO)
 	assert(t, error == nil, "Unable to set log level")
 	tearDown()
 }
@@ -118,14 +117,14 @@ func TestComponent(t *testing.T) {
 func TestInfoCutoff(t *testing.T) {
 	setUp()
 	log := MakeLogger(MockLogger{})
-	log.SetLogLevel(glog.WARNING)
-	assert(t, log.filterLevel == glog.WARNING, "Unable to set log level")
+	log.SetLogLevel(WARNING)
+	assert(t, log.filterLevel == WARNING, "Unable to set log level")
 
-	log = log.Level(glog.INFO)
+	log = log.Level(INFO)
 	log.Log("This is an info message")
 	assert(t, !logCalled, "Info log entry should not have been recorded")
 
-	log = log.Level(glog.WARNING)
+	log = log.Level(WARNING)
 	log.Log("This is a warning message")
 	assert(t, logCalled, "Warning log entry should have been recorded")
 	tearDown()
@@ -181,13 +180,13 @@ func TestCachedLoggers(t *testing.T) {
 	log.SetLogger(logger)
 
 	// set a value on this log class
-	log.SetLogLevel(glog.ERROR)
+	log.SetLogLevel(ERROR)
 	// obtain a new filtered logger and prove it reflects that same log level
 
 	log2 := Logger("test")
 
-	assert(t, log.filterLevel == glog.ERROR, "Log object was not correctly filtered")
-	assert(t, log2.filterLevel == glog.ERROR, "Log object was not cached")
+	assert(t, log.filterLevel == ERROR, "Log object was not correctly filtered")
+	assert(t, log2.filterLevel == ERROR, "Log object was not cached")
 	tearDown()
 }
 
@@ -195,10 +194,10 @@ func TestWarningCutoff(t *testing.T) {
 	setUp()
 	log := MakeLogger(MockLogger{})
 
-	log.Level(glog.WARNING).Log("message", "test warning message")
+	log.Level(WARNING).Log("message", "test warning message")
 	assert(t, logCalled, "Warning level message should have been recorded")
 
-	log.Level(glog.ERROR).Log("error", "test error message")
+	log.Level(ERROR).Log("error", "test error message")
 	assert(t, logCalled, "Error level message should have been recorded")
 	tearDown()
 }
@@ -207,20 +206,20 @@ func TestLogConcurrency(t *testing.T) {
 	setUp()
 	log := MakeLogger(MockLogger{})
 	// create a new log object from the previous one.
-	log2 := log.Level(glog.WARNING)
+	log2 := log.Level(WARNING)
 	assert(t, log.currentLogLevel != log2.currentLogLevel, "log and log2 should not have the same log level")
-	assert(t, log.currentLogLevel == glog.INFO, "Calling Warning() did not create a new log object")
+	assert(t, log.currentLogLevel == INFO, "Calling Warning() did not create a new log object")
 	tearDown()
 }
 
 func TestInfoMessage(t *testing.T) {
 	setUp()
 	log := MakeLogger(MockLogger{})
-	log.SetLogLevel(glog.INFO)
-	log.Level(glog.INFO).Log("test", "message")
+	log.SetLogLevel(INFO)
+	log.Level(INFO).Log("test", "message")
 	logEntry := logParams[0].([]interface{})
 	assert(t, logEntry[0].(string) == "level", "Logged line did not have level entry")
-	assert(t, logEntry[1].(string) == glog.LogLevelNames[glog.INFO], "Logged line was not glog.INFO level")
+	assert(t, logEntry[1].(string) == LogLevelNames[INFO], "Logged line was not infoLevel level")
 	assert(t, logEntry[2].(string) == "timestamp", "Logged line is not expected format")
 	assert(t, logEntry[6].(string) == "component", "Logged line is not expected format")
 	assert(t, logEntry[7].(string) == "test", "Component was not logged")
@@ -230,11 +229,11 @@ func TestInfoMessage(t *testing.T) {
 func TestWarningMessage(t *testing.T) {
 	setUp()
 	log := MakeLogger(MockLogger{})
-	log.SetLogLevel(glog.WARNING)
-	log.Level(glog.WARNING).Log("test", "message")
+	log.SetLogLevel(WARNING)
+	log.Level(WARNING).Log("test", "message")
 	logEntry := logParams[0].([]interface{})
 	assert(t, logEntry[0].(string) == "level", "Logged line did not have level entry")
-	assert(t, logEntry[1].(string) == glog.LogLevelNames[glog.WARNING], "Logged line was not glog.WARNING level")
+	assert(t, logEntry[1].(string) == LogLevelNames[WARNING], "Logged line was not warningLevel level")
 	assert(t, logEntry[2].(string) == "timestamp", "Logged line is not expected format")
 	assert(t, logEntry[6].(string) == "component", "Logged line is not expected format")
 	assert(t, logEntry[7].(string) == "test", "Component was not logged")
@@ -244,11 +243,11 @@ func TestWarningMessage(t *testing.T) {
 func TestErrorMessage(t *testing.T) {
 	setUp()
 	log := MakeLogger(MockLogger{})
-	log.SetLogLevel(glog.ERROR)
-	log.Level(glog.ERROR).Log("test", "message")
+	log.SetLogLevel(ERROR)
+	log.Level(ERROR).Log("test", "message")
 	logEntry := logParams[0].([]interface{})
 	assert(t, logEntry[0].(string) == "level", "Logged line did not have level entry")
-	assert(t, logEntry[1].(string) == glog.LogLevelNames[glog.ERROR], "Logged line was not glog.ERROR level")
+	assert(t, logEntry[1].(string) == LogLevelNames[ERROR], "Logged line was not errorLevel level")
 	assert(t, logEntry[2].(string) == "timestamp", "Logged line is not expected format")
 	assert(t, logEntry[6].(string) == "component", "Logged line is not expected format")
 	assert(t, logEntry[7].(string) == "test", "Component was not logged")
@@ -258,11 +257,11 @@ func TestErrorMessage(t *testing.T) {
 func TestCriticalMessage(t *testing.T) {
 	setUp()
 	log := MakeLogger(MockLogger{})
-	log.SetLogLevel(glog.FATAL)
-	log.Level(glog.FATAL).Log("test", "message")
+	log.SetLogLevel(FATAL)
+	log.Level(FATAL).Log("test", "message")
 	logEntry := logParams[0].([]interface{})
 	assert(t, logEntry[0].(string) == "level", "Logged line did not have level entry")
-	assert(t, logEntry[1].(string) == glog.LogLevelNames[glog.FATAL], "Logged line was not glog.FATAL level")
+	assert(t, logEntry[1].(string) == LogLevelNames[FATAL], "Logged line was not fatalLevel level")
 	assert(t, logEntry[2].(string) == "timestamp", "Logged line is not expected format")
 	assert(t, logEntry[6].(string) == "component", "Logged line is not expected format")
 	assert(t, logEntry[7].(string) == "test", "Component was not logged")
@@ -272,12 +271,12 @@ func TestCriticalMessage(t *testing.T) {
 func TestObject(t *testing.T) {
 	setUp()
 	log := MakeLogger(MockLogger{})
-	log.SetLogLevel(glog.INFO)
+	log.SetLogLevel(INFO)
 	vm := v1.VirtualMachineInstance{ObjectMeta: v12.ObjectMeta{Namespace: "test"}}
 	log.Object(&vm).Log("test", "message")
 	logEntry := logParams[0].([]interface{})
 	assert(t, logEntry[0].(string) == "level", "Logged line did not have level entry")
-	assert(t, logEntry[1].(string) == glog.LogLevelNames[glog.INFO], "Logged line was not of level glog.INFO")
+	assert(t, logEntry[1].(string) == LogLevelNames[INFO], "Logged line was not of level infoLevel")
 	assert(t, logEntry[2].(string) == "timestamp", "Logged line is not expected format")
 	assert(t, logEntry[4].(string) == "pos", "Logged line was not pos")
 	assert(t, logEntry[6].(string) == "component", "Logged line is not expected format")
@@ -292,7 +291,7 @@ func TestObject(t *testing.T) {
 func TestObjectRef(t *testing.T) {
 	setUp()
 	log := MakeLogger(MockLogger{})
-	log.SetLogLevel(glog.INFO)
+	log.SetLogLevel(INFO)
 	vmRef := &k8sv1.ObjectReference{
 		Kind:      "test",
 		Name:      "test",
@@ -302,7 +301,7 @@ func TestObjectRef(t *testing.T) {
 	log.ObjectRef(vmRef).Log("test", "message")
 	logEntry := logParams[0].([]interface{})
 	assert(t, logEntry[0].(string) == "level", "Logged line did not have level entry")
-	assert(t, logEntry[1].(string) == glog.LogLevelNames[glog.INFO], "Logged line was not of level glog.INFO")
+	assert(t, logEntry[1].(string) == LogLevelNames[INFO], "Logged line was not of level infoLevel")
 	assert(t, logEntry[2].(string) == "timestamp", "Logged line is not expected format")
 	assert(t, logEntry[4].(string) == "pos", "Logged line was not pos")
 	assert(t, logEntry[6].(string) == "component", "Logged line is not expected format")
@@ -317,9 +316,9 @@ func TestObjectRef(t *testing.T) {
 func TestError(t *testing.T) {
 	setUp()
 	log := MakeLogger(MockLogger{})
-	log.SetLogLevel(glog.INFO)
+	log.SetLogLevel(INFO)
 	err := errors.New("Test error")
-	log.Level(glog.ERROR).Log(err)
+	log.Level(ERROR).Log(err)
 	assert(t, logCalled, "Error was not logged via .Log()")
 
 	logCalled = false
@@ -340,13 +339,13 @@ func TestError(t *testing.T) {
 func TestMultipleLevels(t *testing.T) {
 	setUp()
 	log := MakeLogger(MockLogger{})
-	log.SetLogLevel(glog.INFO)
+	log.SetLogLevel(INFO)
 	// change levels more than once
-	log.Level(glog.WARNING).Level(glog.INFO).Level(glog.WARNING).msg("test")
+	log.Level(WARNING).Level(INFO).Level(WARNING).msg("test")
 
 	logEntry := logParams[0].([]interface{})
 	assert(t, logEntry[0].(string) == "level", "Logged line did not have level entry")
-	assert(t, logEntry[1].(string) == glog.LogLevelNames[glog.WARNING], "Logged line was not of level glog.WARNING")
+	assert(t, logEntry[1].(string) == LogLevelNames[WARNING], "Logged line was not of level warningLevel")
 	assert(t, logEntry[2].(string) == "timestamp", "Logged line is not expected format")
 	assert(t, logEntry[4].(string) == "pos", "Logged line was not pos")
 	assert(t, logEntry[6].(string) == "component", "Logged line is not expected format")
@@ -359,7 +358,7 @@ func TestMultipleLevels(t *testing.T) {
 func TestLogVerbosity(t *testing.T) {
 	setUp()
 	log := MakeLogger(MockLogger{})
-	log.SetLogLevel(glog.INFO)
+	log.SetLogLevel(INFO)
 	log.SetVerbosityLevel(2)
 	log.V(2).Log("msg", "test")
 
@@ -372,7 +371,7 @@ func TestLogVerbosity(t *testing.T) {
 func TestMsgVerbosity(t *testing.T) {
 	setUp()
 	log := MakeLogger(MockLogger{})
-	log.SetLogLevel(glog.INFO)
+	log.SetLogLevel(INFO)
 	log.SetVerbosityLevel(2)
 	log.V(2).msg("test")
 
@@ -384,7 +383,7 @@ func TestMsgVerbosity(t *testing.T) {
 func TestMsgfVerbosity(t *testing.T) {
 	setUp()
 	log := MakeLogger(MockLogger{})
-	log.SetLogLevel(glog.INFO)
+	log.SetLogLevel(INFO)
 	log.SetVerbosityLevel(2)
 	log.V(2).msgf("%s", "test")
 

--- a/pkg/rest/filter/filter.go
+++ b/pkg/rest/filter/filter.go
@@ -23,7 +23,6 @@ import (
 	"strings"
 
 	"github.com/emicklei/go-restful"
-	"github.com/golang/glog"
 
 	"kubevirt.io/kubevirt/pkg/log"
 )
@@ -37,7 +36,7 @@ func RequestLoggingFilter() restful.FilterFunction {
 			}
 		}
 		chain.ProcessFilter(req, resp)
-		log.Log.Level(glog.INFO).
+		log.Log.Level(log.INFO).
 			With("remoteAddress", strings.Split(req.Request.RemoteAddr, ":")[0]).
 			With("username", username).
 			With("method", req.Request.Method).

--- a/pkg/service/service.go
+++ b/pkg/service/service.go
@@ -47,7 +47,9 @@ func (service *ServiceListen) Address() string {
 }
 
 func (service *ServiceListen) InitFlags() {
-	flag.CommandLine.AddGoFlagSet(goflag.CommandLine)
+	flag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("v"))
+	flag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("kubeconfig"))
+	flag.CommandLine.AddGoFlag(goflag.CommandLine.Lookup("master"))
 }
 
 func (service *ServiceListen) AddCommonFlags() {

--- a/pkg/staging/glog/glog.go
+++ b/pkg/staging/glog/glog.go
@@ -22,7 +22,6 @@ import (
 	"bytes"
 	"flag"
 	"fmt"
-	"io"
 	stdLog "log"
 	"os"
 	"path/filepath"
@@ -35,48 +34,39 @@ import (
 	log2 "github.com/go-kit/kit/log"
 )
 
-type LogLevel int32
+type logLevel int32
 
 const (
-	INFO LogLevel = iota
-	WARNING
-	ERROR
-	FATAL
+	infoLevel logLevel = iota
+	warningLevel
+	errorLevel
+	fatalLevel
 )
 
-var LogLevelNames = map[LogLevel]string{
-	INFO:    "info",
-	WARNING: "warning",
-	ERROR:   "error",
-	FATAL:   "fatal",
+var logLevelNames = map[logLevel]string{
+	infoLevel:    "info",
+	warningLevel: "warning",
+	errorLevel:   "error",
+	fatalLevel:   "fatal",
 }
 
-var logger log2.Logger
-var glogVerbosity int
+var glogVerbosity string
 var glogComponent string
+var toStderr bool
+var logger log2.Logger
 
 func init() {
-	flag.String("v", "2", "log level for V logs")
+	flag.StringVar(&glogVerbosity, "v", "2", "log level for V logs")
+	flag.StringVar(&glogComponent, "component", "", "logger component")
+	flag.BoolVar(&toStderr, "logtostderr", false, "log to standard error instead of files")
 	logger = log2.NewJSONLogger(os.Stderr)
 }
 
-func Setup(component string, verbosity int) {
-	glogVerbosity = verbosity
-	glogComponent = component
-	CopyStandardLogTo(LogLevelNames[INFO])
-}
-
-// SetIOWriter takes a writer to which the logger should write. Useful for redirecting logs during testing.
-func SetIOWriter(w io.Writer) {
-	logger = log2.NewJSONLogger(w)
-	CopyStandardLogTo(LogLevelNames[INFO])
-}
-
-func severityByName(s string) (LogLevel, bool) {
+func severityByName(s string) (logLevel, bool) {
 	s = strings.ToLower(s)
-	for i, name := range LogLevelNames {
+	for i, name := range logLevelNames {
 		if name == s {
-			return LogLevel(i), true
+			return logLevel(i), true
 		}
 	}
 	return 0, false
@@ -158,7 +148,7 @@ func Flush() {
 // severities.  Subsequent changes to the standard log's default output location
 // or format may break this behavior.
 //
-// Valid names are "INFO", "WARNING", "ERROR", and "FATAL".  If the name is not
+// Valid names are "infoLevel", "warningLevel", "errorLevel", and "fatalLevel".  If the name is not
 // recognized, CopyStandardLogTo panics.
 func CopyStandardLogTo(name string) {
 	sev, ok := severityByName(name)
@@ -173,7 +163,7 @@ func CopyStandardLogTo(name string) {
 
 // logBridge provides the Write method that enables CopyStandardLogTo to connect
 // Go's standard logs to the logs provided by this package.
-type logBridge LogLevel
+type logBridge logLevel
 
 // Write parses the standard logging line and passes its components to the
 // logger for severity(lb).
@@ -196,7 +186,7 @@ func (lb logBridge) Write(b []byte) (n int, err error) {
 		}
 	}
 
-	doLogPos(LogLevel(lb), file, line, text)
+	doLogPos(logLevel(lb), file, line, text)
 	return len(b), nil
 }
 
@@ -222,7 +212,10 @@ func V(level Level) Verbose {
 	// This function tries hard to be cheap unless there's work to do.
 	// The fast path is two atomic loads and compares.
 
-	// Here is a cheap but safe test to see if V logging is enabled globally.
+	glogVerbosity, err := strconv.Atoi(glogVerbosity)
+	if err != nil {
+		Fatalf("Verbosity level is invalid: %v", err)
+	}
 	if glogVerbosity >= int(level) {
 		return Verbose(true)
 	}
@@ -233,7 +226,7 @@ func V(level Level) Verbose {
 // See the documentation of V for usage.
 func (v Verbose) Info(args ...interface{}) {
 	if v {
-		doLog(2, INFO, args...)
+		doLog(2, infoLevel, args...)
 	}
 }
 
@@ -241,7 +234,7 @@ func (v Verbose) Info(args ...interface{}) {
 // See the documentation of V for usage.
 func (v Verbose) Infoln(args ...interface{}) {
 	if v {
-		doLog(2, INFO, args...)
+		doLog(2, infoLevel, args...)
 	}
 }
 
@@ -249,110 +242,110 @@ func (v Verbose) Infoln(args ...interface{}) {
 // See the documentation of V for usage.
 func (v Verbose) Infof(format string, args ...interface{}) {
 	if v {
-		doLogf(2, INFO, format, args...)
+		doLogf(2, infoLevel, format, args...)
 	}
 }
 
-// Info logs to the INFO log.
+// Info logs to the infoLevel log.
 // Arguments are handled in the manner of fmt.Print; a newline is appended if missing.
 func Info(args ...interface{}) {
-	doLog(2, INFO, args...)
+	doLog(2, infoLevel, args...)
 }
 
 // InfoDepth acts as Info but uses depth to determine which call frame to log.
 // InfoDepth(0, "msg") is the same as Info("msg").
 func InfoDepth(depth int, args ...interface{}) {
-	doLog(2+depth, INFO, args...)
+	doLog(2+depth, infoLevel, args...)
 }
 
-// Infoln logs to the INFO log.
+// Infoln logs to the infoLevel log.
 // Arguments are handled in the manner of fmt.Println; a newline is appended if missing.
 func Infoln(args ...interface{}) {
-	doLog(2, INFO, args...)
+	doLog(2, infoLevel, args...)
 }
 
-// Infof logs to the INFO log.
+// Infof logs to the infoLevel log.
 // Arguments are handled in the manner of fmt.Printf; a newline is appended if missing.
 func Infof(format string, args ...interface{}) {
-	doLogf(2, INFO, format, args...)
+	doLogf(2, infoLevel, format, args...)
 }
 
-// Warning logs to the WARNING and INFO logs.
+// Warning logs to the warningLevel and infoLevel logs.
 // Arguments are handled in the manner of fmt.Print; a newline is appended if missing.
 func Warning(args ...interface{}) {
-	doLog(2, WARNING, args...)
+	doLog(2, warningLevel, args...)
 }
 
 // WarningDepth acts as Warning but uses depth to determine which call frame to log.
 // WarningDepth(0, "msg") is the same as Warning("msg").
 func WarningDepth(depth int, args ...interface{}) {
-	doLog(2+depth, WARNING, args...)
+	doLog(2+depth, warningLevel, args...)
 }
 
-// Warningln logs to the WARNING and INFO logs.
+// Warningln logs to the warningLevel and infoLevel logs.
 // Arguments are handled in the manner of fmt.Println; a newline is appended if missing.
 func Warningln(args ...interface{}) {
-	doLog(2, WARNING, args...)
+	doLog(2, warningLevel, args...)
 }
 
-// Warningf logs to the WARNING and INFO logs.
+// Warningf logs to the warningLevel and infoLevel logs.
 // Arguments are handled in the manner of fmt.Printf; a newline is appended if missing.
 func Warningf(format string, args ...interface{}) {
-	doLogf(2, WARNING, format, args...)
+	doLogf(2, warningLevel, format, args...)
 }
 
-// Error logs to the ERROR, WARNING, and INFO logs.
+// Error logs to the errorLevel, warningLevel, and infoLevel logs.
 // Arguments are handled in the manner of fmt.Print; a newline is appended if missing.
 func Error(args ...interface{}) {
-	doLog(2, ERROR, args...)
+	doLog(2, errorLevel, args...)
 }
 
 // ErrorDepth acts as Error but uses depth to determine which call frame to log.
 // ErrorDepth(0, "msg") is the same as Error("msg").
 func ErrorDepth(depth int, args ...interface{}) {
-	doLog(2+depth, ERROR, args...)
+	doLog(2+depth, errorLevel, args...)
 }
 
-// Errorln logs to the ERROR, WARNING, and INFO logs.
+// Errorln logs to the errorLevel, warningLevel, and infoLevel logs.
 // Arguments are handled in the manner of fmt.Println; a newline is appended if missing.
 func Errorln(args ...interface{}) {
-	doLog(2, ERROR, args...)
+	doLog(2, errorLevel, args...)
 }
 
-// Errorf logs to the ERROR, WARNING, and INFO logs.
+// Errorf logs to the errorLevel, warningLevel, and infoLevel logs.
 // Arguments are handled in the manner of fmt.Printf; a newline is appended if missing.
 func Errorf(format string, args ...interface{}) {
-	doLogf(2, ERROR, format, args...)
+	doLogf(2, errorLevel, format, args...)
 }
 
-// Fatal logs to the FATAL, ERROR, WARNING, and INFO logs,
+// Fatal logs to the fatalLevel, errorLevel, warningLevel, and infoLevel logs,
 // including a stack trace of all running goroutines, then calls os.Exit(255).
 // Arguments are handled in the manner of fmt.Print; a newline is appended if missing.
 func Fatal(args ...interface{}) {
-	doLog(2, FATAL, args...)
+	doLog(2, fatalLevel, args...)
 	os.Exit(255)
 }
 
 // FatalDepth acts as Fatal but uses depth to determine which call frame to log.
 // FatalDepth(0, "msg") is the same as Fatal("msg").
 func FatalDepth(depth int, args ...interface{}) {
-	doLog(2+depth, FATAL, args...)
+	doLog(2+depth, fatalLevel, args...)
 	os.Exit(255)
 }
 
-// Fatalln logs to the FATAL, ERROR, WARNING, and INFO logs,
+// Fatalln logs to the fatalLevel, errorLevel, warningLevel, and infoLevel logs,
 // including a stack trace of all running goroutines, then calls os.Exit(255).
 // Arguments are handled in the manner of fmt.Println; a newline is appended if missing.
 func Fatalln(args ...interface{}) {
-	doLog(2, FATAL, args...)
+	doLog(2, fatalLevel, args...)
 	os.Exit(255)
 }
 
-// Fatalf logs to the FATAL, ERROR, WARNING, and INFO logs,
+// Fatalf logs to the fatalLevel, errorLevel, warningLevel, and infoLevel logs,
 // including a stack trace of all running goroutines, then calls os.Exit(255).
 // Arguments are handled in the manner of fmt.Printf; a newline is appended if missing.
 func Fatalf(format string, args ...interface{}) {
-	doLogf(2, FATAL, format, args...)
+	doLogf(2, fatalLevel, format, args...)
 	os.Exit(255)
 }
 
@@ -360,38 +353,41 @@ func Fatalf(format string, args ...interface{}) {
 // It allows Exit and relatives to use the Fatal logs.
 var fatalNoStacks uint32
 
-// Exit logs to the FATAL, ERROR, WARNING, and INFO logs, then calls os.Exit(1).
+// Exit logs to the fatalLevel, errorLevel, warningLevel, and infoLevel logs, then calls os.Exit(1).
 // Arguments are handled in the manner of fmt.Print; a newline is appended if missing.
 func Exit(args ...interface{}) {
-	doLog(2, FATAL, args...)
+	doLog(2, fatalLevel, args...)
 	os.Exit(1)
 }
 
 // ExitDepth acts as Exit but uses depth to determine which call frame to log.
 // ExitDepth(0, "msg") is the same as Exit("msg").
 func ExitDepth(depth int, args ...interface{}) {
-	doLog(2+depth, FATAL, args)
+	doLog(2+depth, fatalLevel, args)
 	os.Exit(1)
 }
 
-// Exitln logs to the FATAL, ERROR, WARNING, and INFO logs, then calls os.Exit(1).
+// Exitln logs to the fatalLevel, errorLevel, warningLevel, and infoLevel logs, then calls os.Exit(1).
 func Exitln(args ...interface{}) {
-	doLog(2, FATAL, args...)
+	doLog(2, fatalLevel, args...)
 	os.Exit(1)
 }
 
-// Exitf logs to the FATAL, ERROR, WARNING, and INFO logs, then calls os.Exit(1).
+// Exitf logs to the fatalLevel, errorLevel, warningLevel, and infoLevel logs, then calls os.Exit(1).
 // Arguments are handled in the manner of fmt.Printf; a newline is appended if missing.
 func Exitf(format string, args ...interface{}) {
-	doLogf(2, FATAL, format, args...)
+	doLogf(2, fatalLevel, format, args...)
 	os.Exit(1)
 }
 
-func doLogf(skipFrames int, severity LogLevel, format string, args ...interface{}) {
+func doLogf(skipFrames int, severity logLevel, format string, args ...interface{}) {
+	if !toStderr {
+		return
+	}
 	now := time.Now()
 	_, fileName, lineNumber, _ := runtime.Caller(skipFrames)
 	logger.Log(
-		"level", LogLevelNames[severity],
+		"level", logLevelNames[severity],
 		"timestamp", now.Format("2006-01-02T15:04:05.000000Z"),
 		"pos", fmt.Sprintf("%s:%d", filepath.Base(fileName), lineNumber),
 		"component", glogComponent,
@@ -399,11 +395,14 @@ func doLogf(skipFrames int, severity LogLevel, format string, args ...interface{
 	)
 }
 
-func doLog(skipFrames int, severity LogLevel, args ...interface{}) {
+func doLog(skipFrames int, severity logLevel, args ...interface{}) {
+	if !toStderr {
+		return
+	}
 	now := time.Now()
 	_, fileName, lineNumber, _ := runtime.Caller(skipFrames)
 	logger.Log(
-		"level", LogLevelNames[severity],
+		"level", logLevelNames[severity],
 		"timestamp", now.Format("2006-01-02T15:04:05.000000Z"),
 		"pos", fmt.Sprintf("%s:%d", filepath.Base(fileName), lineNumber),
 		"component", glogComponent,
@@ -411,10 +410,13 @@ func doLog(skipFrames int, severity LogLevel, args ...interface{}) {
 	)
 }
 
-func doLogPos(severity LogLevel, fileName string, lineNumber int, args ...interface{}) {
+func doLogPos(severity logLevel, fileName string, lineNumber int, args ...interface{}) {
+	if !toStderr {
+		return
+	}
 	now := time.Now()
 	logger.Log(
-		"level", LogLevelNames[severity],
+		"level", logLevelNames[severity],
 		"timestamp", now.Format("2006-01-02T15:04:05.000000Z"),
 		"pos", fmt.Sprintf("%s:%d", filepath.Base(fileName), lineNumber),
 		"component", glogComponent,

--- a/pkg/virt-controller/leaderelectionconfig/config.go
+++ b/pkg/virt-controller/leaderelectionconfig/config.go
@@ -20,9 +20,9 @@
 package leaderelectionconfig
 
 import (
-	"flag"
 	"time"
 
+	"github.com/spf13/pflag"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	rl "k8s.io/client-go/tools/leaderelection/resourcelock"
 )
@@ -46,20 +46,20 @@ func DefaultLeaderElectionConfiguration() Configuration {
 
 // BindFlags binds the common LeaderElectionCLIConfig flags
 func BindFlags(l *Configuration) {
-	flag.DurationVar(&l.LeaseDuration.Duration, "leader-elect-lease-duration", l.LeaseDuration.Duration, ""+
+	pflag.DurationVar(&l.LeaseDuration.Duration, "leader-elect-lease-duration", l.LeaseDuration.Duration, ""+
 		"The duration that non-leader candidates will wait after observing a leadership "+
 		"renewal until attempting to acquire leadership of a led but unrenewed leader "+
 		"slot. This is effectively the maximum duration that a leader can be stopped "+
 		"before it is replaced by another candidate. This is only applicable if leader "+
 		"election is enabled.")
-	flag.DurationVar(&l.RenewDeadline.Duration, "leader-elect-renew-deadline", l.RenewDeadline.Duration, ""+
+	pflag.DurationVar(&l.RenewDeadline.Duration, "leader-elect-renew-deadline", l.RenewDeadline.Duration, ""+
 		"The interval between attempts by the acting master to renew a leadership slot "+
 		"before it stops leading. This must be less than or equal to the lease duration. "+
 		"This is only applicable if leader election is enabled.")
-	flag.DurationVar(&l.RetryPeriod.Duration, "leader-elect-retry-period", l.RetryPeriod.Duration, ""+
+	pflag.DurationVar(&l.RetryPeriod.Duration, "leader-elect-retry-period", l.RetryPeriod.Duration, ""+
 		"The duration the clients should wait between attempting acquisition and renewal "+
 		"of a leadership. This is only applicable if leader election is enabled.")
-	flag.StringVar(&l.ResourceLock, "leader-elect-resource-lock", l.ResourceLock, ""+
+	pflag.StringVar(&l.ResourceLock, "leader-elect-resource-lock", l.ResourceLock, ""+
 		"The type of resource object that is used for locking during "+
 		"leader election. Supported options are `endpoints` (default) and `configmap`.")
 }

--- a/pkg/virt-controller/watch/application.go
+++ b/pkg/virt-controller/watch/application.go
@@ -116,11 +116,11 @@ func Execute() {
 	var err error
 	var app VirtControllerApp = VirtControllerApp{}
 
-	featuregates.ParseFeatureGatesFromConfigMap()
-
 	app.LeaderElection = leaderelectionconfig.DefaultLeaderElectionConfiguration()
 
 	service.Setup(&app)
+
+	featuregates.ParseFeatureGatesFromConfigMap()
 
 	app.readyChan = make(chan bool, 1)
 
@@ -198,7 +198,7 @@ func (vca *VirtControllerApp) Run() {
 	defer close(stop)
 	go func() {
 		httpLogger := logger.With("service", "http")
-		httpLogger.Level(glog.INFO).Log("action", "listening", "interface", vca.BindAddress, "port", vca.Port)
+		httpLogger.Level(log.INFO).Log("action", "listening", "interface", vca.BindAddress, "port", vca.Port)
 		http.Handle("/metrics", promhttp.Handler())
 		if err := http.ListenAndServeTLS(vca.Address(), certStore.CurrentPath(), certStore.CurrentPath(), nil); err != nil {
 			golog.Fatal(err)

--- a/pkg/virt-handler/migration-proxy/migration_proxy_suite_test.go
+++ b/pkg/virt-handler/migration-proxy/migration_proxy_suite_test.go
@@ -5,9 +5,12 @@ import (
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+
+	"kubevirt.io/kubevirt/pkg/log"
 )
 
 func TestMigrationProxy(t *testing.T) {
 	RegisterFailHandler(Fail)
+	log.Log.SetIOWriter(GinkgoWriter)
 	RunSpecs(t, "MigrationProxy Suite")
 }

